File: algoexpert/python/medium/validateBst.py
def validateBst(tree):
    def dfs(node, min, max):
        if node == None:
            return True
        if node.value < min or node.value >= max:
            return False

        return dfs(node.left, min, node.value) and dfs(node.right, node.value, max)

    return dfs(tree, float('-inf'), float('inf'))

File: algoexpert/javascript/medium/validateBst.js
function validateBst(tree) {
  function dfs(node, min, max) {
    if (node === null) return true;
    if (node.value < min || node.value >= max) return false;
    return dfs(node.left, min, node.value) && dfs(node.right, node.value, max);
  }
  return dfs(tree, -Infinity, Infinity);
}